File: hw/fuentes/U3/HW20COMPOSITE/HW20COMPOSITEPY/ec/edu/espe/composite/model/President.py
from model.Supervisor import Supervisor

class President(Supervisor):
    __instance = None

    def __new__(cls, aName=""):
        if cls.__instance is None:
            cls.__instance = super(President, cls).__new__(cls)
            cls.__instance.__initialized = False
        return cls.__instance

    def __init__(self, aName=""):
        if self.__initialized:
            self.name = aName
            return
        super().__init__()
        self.title = "President"
        self.name = aName
        President.__instance = self
        self.__initialized = True

    def stateName(self):
        super().stateName()

File: hw/suarez/U3/HW19Composite/src/main/java/ec/edu/espe/composite/model/President.java
package ec.edu.espe.composite.model;

/**
 * @author bernardo
 */
public class President extends Supervisor { //add public

    private static President president = new President();

    private President(String aName) {
        this();
        name = aName;
    }

    private President() {
        super();
        title = "President";
    }

    public void stateName() { // Do processing special to presidential naming
        super.stateName();
    }

    public static President getPresident(String aName) {
        president.name = aName;
        return President.president;
    }
}
